relay/sources/rudder-package/src/archive.rs

```diff
@@ -183,23 +183,21 @@ impl Rpkg {
         Ok(())
     }
 
-    pub fn is_installed(&self) -> Result<bool> {
-        let current_database = Database::read(PACKAGES_DATABASE_PATH)?;
-        Ok(current_database.is_installed(self.to_owned()))
+    pub fn is_installed(&self, db: &Database) -> Result<bool> {
+        Ok(db.is_installed(self.to_owned()))
     }
 
-    pub fn install(&self, force: bool, webapp: &mut Webapp) -> Result<()> {
-        debug!("Installing rpkg '{}'...", self.path);
+    pub fn install(&self, force: bool, db: &mut Database, webapp: &mut Webapp) -> Result<()> {
```

Comment:
We should have an upgrade method, as when upgrading, we should start by removing the installed files and jars from the previous version after running depency check